Various updates to fixes issues

lucasnz opened the pull request:
1. Add Spa manufacturer, controller model, and link to ESP web interface. https://github.com/wayne-love/sn_esp32/issues/55
2. Possible fix for strange times on timers. https://github.com/wayne-love/sn_esp32/issues/35
3. Don't publish entities that aren't connected/supported. https://github.com/wayne-love/sn_esp32/issues/54

wayne-love commented:
We cannot use SpaNet as it is a trademark, we should use SN_ESP32 until we come up with something better.

It would be better to expose the release number of the code rather than the firmware version of the spa.  The firmware version could be exposed as a sensor.

See my comments in discord about using mappings rather than if...then... statements

lucasnz commented:
Lets use SN_ESP32 then. In that case, should the model match the two options in platform.io: esp32dev or spa-control-pcb ??

Let's discuss the inclusions/exclusions based on Spa controller in discord...

lucasnz commented:
I have updated this PR as follows:
1. Inclusion of build version as a macro BUILD_INFO. We use the git tag if it's available (see logic in python script).
2. Mqtt now entity values changed; manufacturer=sn_esp32, model=<platformio env> e.g. esp32dev or spa-control-pcb.
3. Added BUILD_INFO to sw_version on mqtt entities.
4. Changed mqtt filtering to hide heat pump if spa response indicates it's not present. Removed similar logic for current mqtt filtering as this was based on controller model number.

Two other fixes unchanged (time issue) and #49 fix.
